Docker: Permanently restarting - Janus exits with code 132

Hi,

I followed the complete docker installation guide without any errors. After starting all containers, I recognized that the containers are permanently restarting. Especially, the container janus terminates with error 132.

Unfortunately, I am not able to see any logs of any container. I am also not able to enter a bash of a container. All containers are within a restarting loop. 

I am using the following distribution on my Rasp Zero W: Raspbian 10 (buster). I did not modify any the content of the repo.

Any ideas? Thanks in advanced!

BR GL

OK. I managed to build Fruitnanny and the GStreamer Containers. While bulding Janus I get some errors. The command I used was
```
pi@raspberry:/opt/fruitnanny/docker/janus $ docker build -f ./Dockerfile /opt/fruitnanny/
```

There were a lot of errors, but the last one I got was: 
```
[...]
Error: Cannot find module 'minimist'
Require stack:
- /usr/lib/nodejs/rollup/bin/rollup
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:17)
    at Function.Module._load (internal/modules/cjs/loader.js:859:27)
    at Module.require (internal/modules/cjs/loader.js:1028:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/usr/lib/nodejs/rollup/bin/rollup:6:32)
    at Module._compile (internal/modules/cjs/loader.js:1139:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1159:10)
    at Module.load (internal/modules/cjs/loader.js:988:32)
    at Function.Module._load (internal/modules/cjs/loader.js:896:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/usr/lib/nodejs/rollup/bin/rollup' ]
}
make[1]: *** [debian/rules:39: debian/js/janus.js] Error 1
make[1]: Leaving directory '/tmp/janus'
make: *** [debian/rules:64: binary] Error 2
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
The command '/bin/sh -c buildDeps="build-essential git debhelper gobject-introspection gtk-doc-tools libgirepository1.0-dev libglib2.0-dev libgnutls28-dev libgstreamer1.0-dev libgupnp-igd-1.0-dev autotools-dev dh-autoreconf dh-systemd gengetopt libavcodec-dev libavformat-dev libavutil-dev libconfig-dev libcurl4-openssl-dev libcurl4-openssl-dev libjansson-dev liblua5.3-dev libmicrohttpd-dev libnanomsg-dev libogg-dev libopus-dev librabbitmq-dev libre-dev libsofia-sip-ua-dev libsrtp2-dev libssl-dev libsystemd-dev libusrsctp-dev libwebsockets-dev pkg-config rename fakeroot gir1.2-glib-2.0 libgirepository-1.0-1 pandoc brotli doxygen graphviz node-rollup-plugin-replace pigz rollup uglifyjs node-terser uglifyjs.terser"     && runDeps="avahi-daemon libnss-mdns"     && echo 'APT::Install-Recommends "false";' >/etc/apt/apt.conf.d/00recommends     && echo 'APT::Install-Suggests "false";' >>/etc/apt/apt.conf.d/00recommends     && apt-get update     && apt-get install $buildDeps $runDeps     && git clone --depth 1 --single-branch --branch debian/0.1.16-1 https://salsa.debian.org/telepathy-team/libnice.git /tmp/libnice     && cd /tmp/libnice     && dpkg-buildpackage -b --no-sign -rfakeroot     && dpkg -i ../libnice10_0.1.16-1_armhf.deb ../gir1.2-nice-0.1_0.1.16-1_armhf.deb ../libnice-dev_0.1.16-1_armhf.deb     && git clone --depth 1 --single-branch --branch debian/0.7.3-2 https://salsa.debian.org/pkg-voip-team/janus.git  /tmp/janus     && cd /tmp/janus     && dpkg-buildpackage -b --no-sign -rfakeroot     && dpkg -i ../janus_0.7.3-2_armhf.deb ../janus-tools_0.7.3-2_armhf.deb     && cd /     && apt-get --purge remove libnice-dev $buildDeps     && apt-get --purge autoremove     && apt-get clean     && rm -rf /tmp/*     && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 2
```

Any hints on that? Which build command do you use, resp. how do you build?

Thanks in advanced.

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -2,7 +2,7 @@ version: '3'
 services:
   janus:
     image: ivadim/fruitnanny-janus
-    restart: "no"
+    restart: always
     privileged: true
     network_mode: "host"
     volumes:
@@ -16,7 +16,7 @@ services:
 
   gstreamer-video:
     image: ivadim/fruitnanny-gstreamer
-    restart: "no"
+    restart: always
     privileged: true
     network_mode: "host"
     depends_on:
@@ -30,7 +30,7 @@ services:
 
   gstreamer-audio:
     image: ivadim/fruitnanny-gstreamer
-    restart: "no"
+    restart: always
     privileged: true
     network_mode: "host"
     depends_on:
@@ -42,7 +42,7 @@ services:
 
   fruitnanny:
     image: ivadim/fruitnanny-app
-    restart: "no"
+    restart: always
     privileged: true
     volumes:
       - ./fruitnanny_config.js:/opt/fruitnanny/fruitnanny_config.js
@@ -54,8 +54,8 @@ services:
       - 7000
 
   nginx:
-    image: arm32v7/nginx
-    restart: "no"
+    image: arm32v6/nginx:alpine
+    restart: always
     network_mode: "host"
     depends_on:
       - janus
@@ -63,6 +63,7 @@ services:
     volumes:
       - ./configuration/nginx/fruitnanny_http:/etc/nginx/conf.d/default.conf
       - ./configuration/nginx/fruitnanny_https:/etc/nginx/conf.d/default-ssl.conf
+      - ./configuration/nginx/.htpasswd:/etc/nginx/.htpasswd
       - ./configuration/ssl:/etc/fruitnanny-ssl
     ports:
       - "80:80"

--- a/docker/fruitnanny/Dockerfile
+++ b/docker/fruitnanny/Dockerfile
@@ -1,4 +1,4 @@
-FROM balenalib/raspberrypi3-debian-node:12.11-buster
+FROM balenalib/raspberry-pi-node:12.11-buster
 
 RUN buildDeps="build-essential python-dev python3-dev python-setuptools python3-setuptools" \
     && echo 'APT::Install-Recommends "false";' >/etc/apt/apt.conf.d/00recommends \

--- a/docker/gstreamer/Dockerfile
+++ b/docker/gstreamer/Dockerfile
@@ -1,4 +1,4 @@
-FROM balenalib/raspberrypi3-debian:buster
+FROM balenalib/raspberry-pi:buster
 
 RUN echo 'APT::Install-Recommends "false";' >/etc/apt/apt.conf.d/00recommends \
     && echo 'APT::Install-Suggests "false";' >>/etc/apt/apt.conf.d/00recommends \

--- a/docker/gstreamer/Dockerfile
+++ b/docker/gstreamer/Dockerfile
@@ -25,5 +25,5 @@ RUN echo 'APT::Install-Recommends "false";' >/etc/apt/apt.conf.d/00recommends \
     && rm -rf /tmp/*
 
 RUN useradd -rm -d /home/pi -s /bin/bash -g users -G adm,dialout,sudo,audio,video,plugdev,input,netdev -u 1000 pi
-COPY audio_entry.sh /
-COPY audio_stream.sh /
+COPY docker/gstreamer/audio_entry.sh /
+COPY docker/gstreamer/audio_stream.sh /

--- a/docker/janus/Dockerfile
+++ b/docker/janus/Dockerfile
@@ -1,4 +1,4 @@
-FROM balenalib/raspberrypi3-debian
+FROM balenalib/raspberry-pi:buster
 
 RUN buildDeps="build-essential git debhelper gobject-introspection gtk-doc-tools libgirepository1.0-dev libglib2.0-dev libgnutls28-dev libgstreamer1.0-dev libgupnp-igd-1.0-dev autotools-dev dh-autoreconf dh-systemd gengetopt libavcodec-dev libavformat-dev libavutil-dev libconfig-dev libcurl4-openssl-dev libcurl4-openssl-dev libjansson-dev liblua5.3-dev libmicrohttpd-dev libnanomsg-dev libogg-dev libopus-dev librabbitmq-dev libre-dev libsofia-sip-ua-dev libsrtp2-dev libssl-dev libsystemd-dev libusrsctp-dev libwebsockets-dev pkg-config rename fakeroot gir1.2-glib-2.0 libgirepository-1.0-1 pandoc brotli doxygen graphviz node-rollup-plugin-replace pigz rollup uglifyjs node-terser uglifyjs.terser" \
     && runDeps="avahi-daemon libnss-mdns" \

